sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -208,7 +211,10 @@ object StateSchemaCompatibilityChecker {
    * @param stateSchemaVersion - version of the state schema to be used
    * @param extraOptions - any extra options to be passed for StateStoreConf creation
    * @param storeName - optional state store name
-   * @param schemaFilePath - optional schema file path
+   * @param oldSchemaFilePath - optional path to the old schema file. If not provided, will default
```

Comment:
Is it valid for schema version 3 to have None? Otherwise let's also mention here as same as below; Needed for schema version 3.

Reply:
Yeah, we don't expect there to be a schema file if we are in the first run of a new query, right?